### DISCUSSION TITLE
Using ES modules in Jest tests

### DISCUSSION THREAD

**Discussion by mitas1** [vercel/next.js#34589] [Help]
Since the Next.js version 12, there is built-in support for ES Modules. That's cool, but what is the recommended way to use ES modules even in Jest tests? One way of doing it was to specify a custom `"transformIgnorePatterns"` in the config, but if I understand it correctly, according to this comment in built-in Jest configuration: https://github.com/vercel/next.js/blob/a74af1f31d6b6be84350679166eb4a236f1186a3/packages/next/build/jest/jest.ts#L122-L123
for some reason, we cannot use `transformIgnorePatterns` for this anymore. Or am I missing something?

**Comment by benj-dobs** [comment]
As a workaround, you can modify the config returned by `next/jest`:

```js
// Based on https://nextjs.org/docs/testing#setting-up-jest-with-the-rust-compiler
// jest.config.js
const nextJest = require('next/jest')

const createJestConfig = nextJest({
  // Provide the path to your Next.js app to load next.config.js and .env files in your test environment
  dir: './',
})

// Add any custom config to be passed to Jest
const customJestConfig = {
  // Add more setup options before each test is run
  // setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
  // if using TypeScript with a baseUrl set to the root directory then you need the below for alias' to work
  moduleDirectories: ['node_modules', '<rootDir>/'],
  testEnvironment: 'jest-environment-jsdom',
}

// createJestConfig is exported this way to ensure that next/jest can load the Next.js config which is async
const getConfig = createJestConfig(customJestConfig)

module.exports = async () => {
  const nextJestConfig = await getConfig()
  const transformIgnorePatterns = [
    '/node_modules/(?!exception_one|exception_two|exception_three)',
    ...nextJestConfig.transformIgnorePatterns.filter(
      (pattern) => pattern !== '/node_modules/'
    )
  ]
}
```

**Reply by jj-ws** [reply]
That works well for me. Just needed to add to the exported function:
```
  return {
    ...nextJestConfig,
    transformIgnorePatterns,
  };
```

It would be good to have this solution in the Jest documentation instead of [the current one](https://jestjs.io/docs/ecmascript-modules).
